modules/mediawiki/manifests/deploy.pp

```diff
@@ -0,0 +1,140 @@
+        owner     => 'www-data',
+        group     => 'www-data',
+        mode      => '0755',
+        require   => File['/srv/mediawiki-staging'],
+    }
+
+    git::clone { 'MediaWiki core':
+        ensure             => 'present',
+        directory          => '/srv/mediawiki-staging/w',
+        origin             => 'https://github.com/miraheze/mediawiki.git',
+        branch             => $branch,
+        owner              => 'www-data',
+        group              => 'www-data',
+        mode               => '0755',
+        timeout            => '1500',
+        depth              => '5',
+        recurse_submodules => true,
+        require            => File['/srv/mediawiki-staging'],
+    }
+
+    git::clone { 'landing':
+        ensure    => 'latest',
+        directory => '/srv/mediawiki-staging/landing',
+        origin    => 'https://github.com/miraheze/landing.git',
+        branch    => 'master',
+        owner     => 'www-data',
+        group     => 'www-data',
+        mode      => '0755',
+        require   => File['/srv/mediawiki-staging'],
+    }
+
+    git::clone { 'ErrorPages':
+        ensure    => 'latest',
+        directory => '/srv/mediawiki-staging/ErrorPages',
+        origin    => 'https://github.com/miraheze/ErrorPages.git',
+        branch    => 'master',
+        owner     => 'www-data',
+        group     => 'www-data',
+        mode      => '0755',
+        require   => File['/srv/mediawiki-staging'],
+    }
+
+    file { '/usr/local/bin/deploy-mediawiki':
+        ensure => 'present',
+        mode   => '0755',
+        source => 'puppet:///modules/mediawiki/bin/deploy-mediawiki.py',
+    }
+
+    file { '/usr/local/bin/mwupgradetool':
+        ensure => 'present',
+        mode   => '0755',
+        source => 'puppet:///modules/mediawiki/bin/mwupgradetool.py',
+    }
+
+    exec { 'MediaWiki Config Sync':
+        command     => "/usr/local/bin/deploy-mediawiki --config --servers=${lookup(mediawiki::default_sync)}",
+        cwd         => '/srv/mediawiki-staging',
+        refreshonly => true,
+        user        => www-data,
+        subscribe   => Git::Clone['MediaWiki config'],
+        require     => File['/srv/mediawiki'],
+    }
+
+    exec { 'Landing Sync':
+        command     => "/usr/local/bin/deploy-mediawiki --landing --servers=${lookup(mediawiki::default_sync)} --no-log",
+        cwd         => '/srv/mediawiki-staging',
+        refreshonly => true,
+        user        => www-data,
+        subscribe   => Git::Clone['landing'],
+        require     => File['/srv/mediawiki'],
+    }
+
+    exec { 'ErrorPages Sync':
+        command     => "/usr/local/bin/deploy-mediawiki --errorpages --servers=${lookup(mediawiki::default_sync)} --no-log",
+        cwd         => '/srv/mediawiki-staging',
+        refreshonly => true,
+        user        => www-data,
+        subscribe   => Git::Clone['ErrorPages'],
+        require     => File['/srv/mediawiki'],
```

Comment:
Also the ssh keys

Reply:
But that wouldn't work on non-canary servers (test101), would it?